test/functional/p2p_filter.py

```diff
@@ -125,8 +143,11 @@ def run_test(self):
 
         self.log.info('Check that we receive a tx in reply to a mempool msg if the filter matches a mempool tx')
         filter_node.merkleblock_received = False
+        filter_node.tx_received = False
         txid = self.nodes[0].sendtoaddress(filter_address, 90)
+        self.nodes[0].p2p.send_message(msg_mempool())
```

Comment:
So this change needs to go

Reply:
I removed it altogether, since receiving tx behavior for this reason is already tested and `msg_mempool` is added.